docs/source/cluster_management/kvs-migration.rst

```diff
@@ -7,54 +7,130 @@ KVS Migration
+
+Running a KVS Migration
+=======================
+
+Preconditions
+-------------
+
+.. danger::
 
-- Targeted sweep tables, as specified in ``TargetedSweepSchema``, none of which are transactional.
+    Read this section carefully before performing any migrations! The KVS migration CLI makes some assumptions;
+    failure to ensure that these assumptions are valid may result in **SEVERE DATA CORRUPTION**.
 
-- Migration checkpoint table, which is a special table used for checkpointing during the migration.
+While KVS migration is running, the service that is using AtlasDB must continuously be offline; if the service is
+running, either some data may be written that will not be migrated over (if the service runs with the source KVS), or
+some data that should be present may not be visible (if the service runs with the destination KVS).
 
-None of the above tables can be migrated, because the migration uses transactional reads and writes.
-Transactional reads will generally skip over any data in the tables above, but it is possible that some of the data would be successfully read and then almost certainly incorrectly migrated.
-Due to legacy reasons, we still drop and create some of the non-transactional tables as described in more detail below.
-The migration is performed in three steps which must be run in order: setup, followed by migrate, and finally validate.
-If the migration needs to be restarted from scratch, running setup again will reset any previous migration state and allow a fresh migration.
+If you are using TimeLock, the TimeLock server must be running in order to do the migration, since we need to
+obtain migration timestamps.
 
-The first step in running any of the three commands is to take out a fresh timestamp in the source KVS, fast-forward the target KVS to a larger timestamp, and then take out two fresh timestamps on the target KVS for the migration start timestamp and the migration commit timestamp.
-Then, we immediately insert an entry into the transactions table of the target KVS with the above two timestamps, effectively committing all transactions with the migration start timestamp.
+If you are using embedded services, the ``--offline`` flag must be used. This will remove the leader block from your
+configuration for the purposes of migration.
 
-.. warning::
+Migration CLI
+-------------
 
-    KVS migration must be run while the service using AtlasDB is offline.
-    If you are using TimeLock, the TimeLock server must be running in order to do the migration.
-    Otherwise, you must use the ``--offline`` flag, which will remove the leader block from your configuration for the purposes of migration.
+.. tip::
+
+    If you are migrating to/from Oracle KVS, the standard AtlasDB CLI distribution will not work as it does not
+    contain the requisite Oracle drivers. Internal users should contact the AtlasDB team for assistance; external
+    users should compile the CLI from source, suitably adding the Oracle drivers.
```

Comment:
it's not sufficient to add the oracle drivers, there is atlasdb oracle config in the internal dist, so external users can *never* compile this more or less